eureka-client-archaius2/src/main/java/com/netflix/appinfo/Ec2EurekaArchaius2InstanceConfig.java

```diff
@@ -41,8 +41,8 @@ public Ec2EurekaArchaius2InstanceConfig(Config configInstance, AmazonInfoConfig
         this(configInstance, new Archaius2AmazonInfoConfig(configInstance), CommonConstants.DEFAULT_CONFIG_NAMESPACE, info, false);
     }
 
-    public Ec2EurekaArchaius2InstanceConfig(Config configInstance, Provider<AmazonInfo> amazonInfoProvider) {
-        super(configInstance, CommonConstants.DEFAULT_CONFIG_NAMESPACE);
+    public Ec2EurekaArchaius2InstanceConfig(Config configInstance, Provider<AmazonInfo> amazonInfoProvider, String namespace) {
```

Comment:
What about a new constructor while the old one preserves the existing behavior?

Reply:
From a pure backward compatibility perspective that's totally the right thing to do. I doubt this gets used by anybody outside of Netflix though. So is it really worth making the code harder to follow?